### ISSUE TITLE
'Fragment' cannot be used as a JSX component

### ISSUE THREAD

**Issue by adarah** [honojs/hono#1402]
### What version of Hono are you using?

3.5.7

### What runtime/platform is your app running on?

None, typescript issue.

### What steps can reproduce the bug?

1. Clone https://github.com/honojs/examples
2. `cd jsx-ssr && npm i`
3. Open `top.tsx`, try placing a `hono/jsx` Fragment as such:
```tsx
import { Fragment } from "hono/jsx";
import { Layout } from "../components/Layout";
import type { Post } from "../index";

const List = (props: { post: Post }) => (
  <li>
    <a href={`/post/${props.post.id}`}>{props.post.title}</a>
  </li>
);

export const Top = (props: { posts: Post[] }) => {
  return (
    <Layout title={"Top"}>
      <main>
        <h2>Posts</h2>
        <ul>
          {props.posts.map((post) => (
            <List post={post} />
          ))}
        </ul>
        <Fragment>
          <p>Hello world</p>
        </Fragment>
      </main>
    </Layout>
  );
};

```

### What is the expected behavior?

Placing Fragments inside JSX should not result in type errors.

### What do you see instead?

<img width="990" alt="image" src="https://github.com/honojs/hono/assets/36046024/070d8a35-8bf4-40a6-b0e4-80625a4a3be8">

Happens on vscode and vim.

### Additional information

This also happens in another project of mine, but I used the examples repo for easier reproduction.

**Comment by msonnemans** [comment]
I am experiencing a similar issue.

Hono: ^3.5.4, 
tsconfig:
`"jsx": "react-jsx",
    "jsxImportSource": "hono/jsx"`

'Fragment cannot be used as a JSX component. [...] Type 'JSXNode' is not assignable to type 'string'.

**Comment by yusukebe** [comment]
Hi @adarah , @msonnemans !

This issue may be fixed with new version v3.5.8. Try it!
